src/core/src/core_logic/PatchInstaller.py

```diff
@@ -283,7 +333,142 @@ def install_updates(self, maintenance_window, package_manager, simulate=False):
             self.status_handler.add_error_to_status(message, Constants.PatchOperationErrorCodes.OPERATION_FAILED)
             self.composite_logger.log_error(message)
 
-        return installed_update_count, patch_installation_successful, maintenance_window_exceeded
+        return installed_update_count
+
+    def include_dependencies(self, package_manager, packages_in_batch, all_packages, packages, package_versions, package_and_dependencies, package_and_dependency_versions):
+        dependencies = package_manager.get_dependent_list(package_and_dependencies)
+
+        for package in dependencies:
+            if package not in all_packages:
+                continue
+            package_and_dependencies.append(package)
+            version = package_versions[packages.index(package)] if package in packages else Constants.DEFAULT_UNSPECIFIED_VALUE
+            package_and_dependency_versions.append(version)
+
+        for package in packages_in_batch:
+            package_manager.add_arch_dependencies(package_manager, package, packages, package_versions, package_and_dependencies, package_and_dependency_versions)
+
+        package_and_dependencies, package_and_dependency_versions = package_manager.dedupe_update_packages(package_and_dependencies, package_and_dependency_versions)
+
+        self.composite_logger.log("Packages including dependencies are: " + str(package_and_dependencies))
+
+    def install_packages_in_batches(self, all_packages, packages, package_versions, maintenance_window, package_manager, simulate=False):
+        number_of_batches = int(math.ceil(len(packages) / float(Constants.MAX_BATCH_SIZE_FOR_PACKAGES)))
+        self.composite_logger.log("\nNumber of packages to be installed: " + str(len(packages)) + "\nBatch Size: " + str(Constants.MAX_BATCH_SIZE_FOR_PACKAGES) + "\nNumber of batches: " + str(number_of_batches))
```

Comment:
Add to log: "Dividing package install in batches. Number of packages to install..." (.... - the log you have right now)